python/cxx/schedule.cpp

```diff
@@ -56,6 +58,57 @@ namespace {
         throw py::key_error( name );
     }
 
+    std::map<std::string, double> get_production_properties(
+        const Schedule& sch, const std::string& name, const size_t& timestep)
+    {
+        const Well* well = nullptr;
+        try{
+            well = &(sch.getWell( name, timestep ));
+        } catch( const std::invalid_argument& e ) {
+            throw py::key_error( fmt::format("well {} is not defined", name ));
+        }
+        if (well->isProducer()) {
+            std::map<std::string, double> props;
+            auto& prod_prop = well->getProductionProperties();
+            {
+                auto rate = prod_prop.OilRate.get<double>();
+                props.insert(std::make_pair("oil_rate", rate));
+            }
+            {
+                auto rate = prod_prop.GasRate.get<double>();
+                props.insert(std::make_pair("gas_rate", rate));
+            }
+            {
+                auto rate = prod_prop.WaterRate.get<double>();
+                props.insert(std::make_pair("water_rate", rate));
+            }
+            {
+                auto rate = prod_prop.LiquidRate.get<double>();
+                props.insert(std::make_pair("liquid_rate", rate));
+            }
+            {
+                auto rate = prod_prop.ResVRate.get<double>();
+                props.insert(std::make_pair("resv_rate", rate));
+            }
+            {
+                auto rate = prod_prop.BHPTarget.get<double>();
+                props.insert(std::make_pair("bhp_target", rate));
+            }
+            {
+                auto rate = prod_prop.THPTarget.get<double>();
+                props.insert(std::make_pair("thp_target", rate));
+            }
+            {
+                auto rate = prod_prop.ALQValue.get<double>();
+                props.insert(std::make_pair("alq_value", rate));
+            }
+            return props;
```

Comment:
Any reason not to write this as
```C++
return {
        { "oil_rate", prod_prop.OilRate.get<double>() },
        { "gas_rate", prod_prop.GasRate.get<double>() },
        // ...
        { "alq_value", prod_prop.ALQValue.get<double>() },
    };
```

Reply:
@bska Thanks that looks much nicer yes! I have updated the PR.